rfc/0012-invalid-values/README.md

```diff
@@ -0,0 +1,110 @@
+<!--
+SPDX-FileCopyrightText: 2023 Friedrich-Alexander-Universitat Erlangen-Nurnberg
+
+SPDX-License-Identifier: AGPL-3.0-only
+-->
+
+# RFC 0012: Invalid Values
+| | |
+|---|---|
+| Feature Tag | `invalid values` |
+| Status | `DISCUSSION` | <!-- Possible values: DRAFT, DISCUSSION, ACCEPTED, REJECTED -->
+| Responsible | `georg-schwarz` | 
+<!-- 
+  Status Overview:
+  - DRAFT: The RFC is not ready for a review and currently under change. Feel free to already ask for feedback on the structure and contents at this stage.
+  - DISCUSSION: The RFC is open for discussion. Usually, we open a PR to trigger discussions.
+  - ACCEPTED: The RFC was accepted. Create issues to prepare implementation of the RFC.
+  - REJECTED: The RFC was rejected. If another revision emerges, switch to status DRAFT.
+-->
+
+## Summary
+This RFC introduces the concept of invalid values as first-class citizen to Jayvee. We will allow more fine-grained handling of invalid values so that users can decide how to proceed with them.
+
+
+## Motivation
+Right now, we just drop invalid rows when transforming to a table and log a warning. Users, however, might instead want to preserve the valid parts of the rows or curate the invalid fields (e.g., by using default values).
+
+
+## Explanation
+We introduce another primitive value type called `INVALID`.
+
+Values that fail complying with their intended value type are assigned to this `INVALID` value type. The value that this value type stores 
+* a text description on how the invalid value emerged (like a log message)
+* a last valid value before turning invalid.
+
+Any kind of transformation via expressions including an `INVALID` value produces an `INVALID` value again. 
+
+### INVALID value handling blocks
+
+#### Block-Type `InvalidTableRowFilter`
+This block-type implements the current default behavior: dropping all rows that have invalid values in them. No configuration parameters are available.
+```
+block MyInvalidRowFilter oftype InvalidTableRowFilter {}
+```
+
+#### Block-Type `InvalidTableCellReplacer`
+This block-type allows replacing all invalid values of a column with a valid value.
+```
+block MyInvalidTemperatureReplacer oftype InvalidTableCellReplacer {
+  column: "Temperature";  // refers to table column
+  value: 21;              // expression producing valid value
```

Comment:
I like this, in the future expressions should be able to access the whole table (e.g. have a value that is the average of the column or computed from other values in the table). That can be in expression extensions though, not in this RFC.